database/src/migrations/20240119000000_taxonomy_tables.ts

```diff
@@ -0,0 +1,102 @@
+import { Knex } from 'knex';
+
+/**
+ * Add tables:
+ * - taxon
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+    ----------------------------------------------------------------------------------------
+    -- Create tables
+    ----------------------------------------------------------------------------------------
+    set search_path=biohub,public;
+
+    CREATE TABLE taxon (
+      taxon_id                integer             GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+      itis_tsn                integer             NOT NULL,
+      bc_taxon_code           varchar(10),
+      scientific_name         varchar(50)         NOT NULL,
+      common_name             varchar(50),
+      itis_data               jsonb               NOT NULL,
+      itis_update_date        timestamptz(6)      NOT NULL,
+      record_effective_date   date                DEFAULT now() NOT NULL,
+      record_end_date         date,
+      create_date             timestamptz(6)      DEFAULT now() NOT NULL,
+      create_user             integer             NOT NULL,
+      update_date             timestamptz(6),
+      update_user             integer,
+      revision_count          integer             DEFAULT 0 NOT NULL,
+
+      CONSTRAINT              taxon_pk            PRIMARY KEY (taxon_id)
+    );
+
+    COMMENT ON COLUMN taxon.taxon_id              IS 'System generated surrogate primary key identifier.';
+    COMMENT ON COLUMN taxon.itis_tsn              IS 'ITIS primary key identifier, populated from ITIS response. ITIS (Integrated Taxonomic Information System), TSN (Taxonomic Serial Number). https://itis.gov/pdf/faq_itis_tsn.pdf';
+    COMMENT ON COLUMN taxon.bc_taxon_code         IS 'British Columbia standard taxon identifier.';
+    COMMENT ON COLUMN taxon.scientific_name       IS 'Taxon scientific name, initially populated from ITIS response.';
```

Comment:
Not 100% certain, but I feel like the `scientific_name` should be `itis_scientific_name` (just implying that it is always the itis value, and not just initially populated from it and then potentially overwritten by BC later). Possibly the same for the `common_name` column? 

Just thinking that we probably want the itis values to be the source of truth at all times, and the aliases just a way of finding the correct itis record.

Thoughts in general?